plugins/destination/duckdb/client/transform_gotype.go

```diff
@@ -0,0 +1,115 @@
+package client
+
+import (
+	"database/sql/driver"
+	"time"
+
+	"github.com/apache/arrow/go/v15/arrow"
+	"github.com/apache/arrow/go/v15/arrow/array"
+	"github.com/cloudquery/plugin-sdk/v4/types"
+	"github.com/marcboeker/go-duckdb"
+)
+
+func getValue(arr arrow.Array, i int) any {
+	if !arr.IsValid(i) {
+		if arrow.TypeEqual(arr.DataType(), types.NewUUIDType()) {
+			return duckdb.UUID{}
+		}
+
+		// return typed nil
+		switch arr.DataType().ID() {
```

Comment:
IMO it's better to do a type switch instead (see ch dst):
```go
switch dt := arr.DataType().(type)
```
with stuff like `*arrow.TImestampType`